config/formats.js

```diff
@@ -801,9 +801,12 @@ let Formats = [
 			pokemon.hp = pokemon.maxhp = template.maxHP = allies[2].maxhp;
 			pokemon.set.evs = allies[2].set.evs;
 			pokemon.set.nature = allies[2].getNature().name;
+			// @ts-ignore
 			pokemon.set.ivs = pokemon.baseIvs = allies[2].set.ivs;
+			// @ts-ignore
 			pokemon.hpType = pokemon.baseHpType = allies[2].baseHpType;
```

Comment:
I don't know if I approve of things like `hpType` and `level` being read-only. If the engine supports them being modified, and OMs want to modify them, this is probably fine?

Reply:
Hm, this is a complicated decision. I don't think I have hard rules here. I guess it's probably fine to make them read-only.

Reply:
It's up to you. I'm perfectly fine changing them back. Though, its pretty trivial to change back in the future as well - its usually much easier making `readonly` things mutable than vice versa.

Reply:
Yeah, I'm fine leaving them read-only if that's more consistent.